### PR TITLE
Add auto generated query tests for tpch

### PR DESCRIPTION
**Pull request by saurabhd336**
Adding the initial set of changes to run auto generated queries against the tpch dataset. By default the test are disabled.

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/DocIdDictionary.java [hunk 1]
```diff
@@ -45,6 +45,11 @@ public int insertionIndexOf(String stringValue) {
     }
   }
 
+  @Override
+  public int indexOf(int value) {
```
Comment (saurabhd336) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/DocIdDictionary.java:49]: This change is needed to be able to support `$docId IN` queries

**Comment by saurabhd336** [comment]
@xiangfu0 @zhtaoxiang for reviews

**Review by xiangfu0 — APPROVED** [review]

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/tpch/generator/TPCHQueryGeneratorV2.java [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+      for (Table table : tables) {
+        getRandomOrderBys(table, tableWiseGroupByCols.get(table.getTableName())).forEach(
+            querySkeleton::addOrderByColumn);
+      }
+    }
+
+    return querySkeleton.toString();
+  }
+
+  public String generateRandomQuery() {
```
Comment (walterddr) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/tpch/generator/TPCHQueryGeneratorV2.java:531]: did we come up with this structure from some reference database or this is purely just coming from us?

- it seems like we didn't have distinct (or group by without agg) examples?
- i saw group-by after join, is there any join after group-by queries? 
- window function is not covered i suppose

did we have a follow up on what needs to be added next?

Comment (saurabhd336) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/tpch/generator/TPCHQueryGeneratorV2.java:531]: @walterddr , this was entirely coming as an extension of https://github.com/apache/pinot/blob/master/pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/QueryGenerator.java#L67, but with support for joins, aggregations, group bys etc.

I can work with you to identify follow ups for this. I think majorly we're missing nested queries, exhaustive set of aggregations, window function etc.

**Pull request merged**